src/client.ts

```diff
@@ -456,4 +468,21 @@ export class RSPClient {
     removeAllListeners(eventName: string): void {
         this.emitter.removeAllListeners(eventName);
     }
+
+    /**
+     * register client capabilities to the server
+     *
+     * @param capabilities client capabilities to register
+     * @param timeout timeout in milliseconds
+     */
+    registerClientCapabilities(capabilities: any, timeout: number = 2000): Promise<Protocol.ServerCapabilitesResponse> {
+        return this.capabilitiesUtil.registerClientCapabilities(capabilities);
+    }
+
+    /**
+     * Returns the capabilities implemented by the client
+     */
+    getCapabilities(): {} {
+        return {'protocol.version': '0.10.0', 'prompt.string': true};
```

Comment:
Again, the capabilities should be typed to match whatever you'd put in the register capabilities request. That way it wouldn't allow you to put a string, boolean pair into a string, string map.